docker/entrypoint.sh

```diff
@@ -162,79 +228,77 @@ cat /tmp/certbot-hosts.txt | xargs -n 1 -I {} oc process -f /tmp/certbot-route.y
-fi
-
-if [ ! -z "$CERTBOT_SERVER" ]; then
-  CERTBOT_ARGS="${CERTBOT_ARGS} --server ${CERTBOT_SERVER}"
-fi
-
-set -x
-# if there is no certificate issue, request a new one
-if [ ! -f "${CERTBOT_CONFIG_DIR}/live/openshift-route-certs/cert.pem" ]; then
-  certbot --config /tmp/certbot.ini certonly $CERTBOT_ARGS --non-interactive --keep-until-expiring --cert-name 'openshift-route-certs' --expand --standalone -d "$(</tmp/certbot-hosts.csv)"
+# Get certificate(s), either combined or individual
+if [ "${CERTBOT_CERT_PER_HOST}" == "true" ]; then
+  echo "Manage individual certificates for each unique host."  
+  for certbot_host in $(</tmp/certbot-hosts.txt); do
+    getCertificate "${certbot_host}" "${certbot_host}"
+  done
 else
-  # if a certificate already exists, request to renew it
-  certbot --config /tmp/certbot.ini renew $CERTBOT_ARGS --no-random-sleep-on-renew --cert-name 'openshift-route-certs'
-fi
-set +x
+  echo "Managing a single certificate covering all managed hosts."
+  getCertificate "openshift-route-certs" "$(</tmp/certbot-hosts.csv)"
 
-if [ "${CERTBOT_DRY_RUN}" == "true" ]; then
-  exit 0
+  # Re-Map the managed route dictionary so all routes get patched with the combined certificate.
+  for route in "${!managedRoutes[@]}"; do
+    managedRoutes[${route}]="openshift-route-certs"
+  done
 fi
 
-if [ -f $CERTBOT_WORK_DIR/deployed ]; then
+if [ -f ${CERTBOT_WORK_DIR}/deployed ]; then
   echo 'New certificate(s) have been issued'
 else
-  echo 'No certificate(s) have been issued'
+  echo 'No new certificate(s) have been issued'
 fi
 
-# if [ -f $CERTBOT_WORK_DIR/deployed ]; then
-echo 'Updating Routes'
-CERTIFICATE="$(awk '{printf "%s\\n", $0}' $CERTBOT_CONFIG_DIR/live/openshift-route-certs/cert.pem)"
-KEY="$(awk '{printf "%s\\n", $0}' $CERTBOT_CONFIG_DIR/live/openshift-route-certs/privkey.pem)"
-CABUNDLE=$(awk '{printf "%s\\n", $0}' $CERTBOT_CONFIG_DIR/live/openshift-route-certs/fullchain.pem)
+# Patch Routes
+for route in "${!managedRoutes[@]}"; do
+  certificateName=${managedRoutes[${route}]}
+
+  echo "Updating route/${route} with certificate ${certificateName} ..."
+  CERTIFICATE="$(awk '{printf "%s\\n", $0}' ${CERTBOT_CONFIG_DIR}/live/${certificateName}/cert.pem)"
+  KEY="$(awk '{printf "%s\\n", $0}' ${CERTBOT_CONFIG_DIR}/live/${certificateName}/privkey.pem)"
+  CABUNDLE=$(awk '{printf "%s\\n", $0}' ${CERTBOT_CONFIG_DIR}/live/${certificateName}/fullchain.pem)
+
+  # If any of the cert components is blank, then don't run the patch command
+  if [ "${CERTBOT_DRY_RUN}" == "true" ]; then
+    echo "Dry Run - the certificate for route/${route} was not patched."
+  elif [ "${CERTIFICATE}" == "" ] || [ "${KEY}" == "" ] || [ "${CABUNDLE}" == "" ]; then
+    echo "The certificate for route/${route} wasn't created properly so it won't be patched."
+  else
+    oc patch "route/${route}" -p '{"spec":{"tls":{"certificate":"'"${CERTIFICATE}"'","key":"'"${KEY}"'","caCertificate":"'"${CABUNDLE}"'"}}}'
```

Comment:
A success echo message here could be useful, albeit verbose.

Reply:
On success to get log output from the route being patched with the certificate.  Patching is performed whenever the certificate for the route exists, whether or not it was renewed.  So you always get an indication the certificate exists and is being applied.  In the case the certificate has not been renewed (remains the same) the patching operation does not make any changes to the route and indicates (no changes).